### PR TITLE
libpng: update url and use tar.gz instead zip

### PR DESCRIPTION
**Pull request by darcy-shen**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

**Comment by CyaNinix** [comment]
why? this doesn't seem right as the github tags are named with a starting "v"

**Comment by SirLynix** [comment]
why? this doesn't seem right as the github tags are named with a starting "v"

**Comment by SirLynix** [comment]
by the way you don't need to make another pull request to trigger CI again after changing base, just close and reopen the same PR

**Comment by darcy-shen** [comment]
> why? this doesn't seem right as the github tags are named with a starting "v"

Because I want to add this mirror as the alternative url:

http://mirrors.ustc.edu.cn/debian/pool/main/libp/libpng1.6/libpng1.6_1.6.36.orig.tar.xz

For me, downloading from Github is unstable. I think use version without v is a better practice. What's your opnion? @waruqi

**Comment by SirLynix** [comment]
it's not about opinion, xmake uses the version name as git tags if archive download fails.
so with `add_versions("v1.6.40", "hash")` xmake will try to download .tar.gz and then fallback on the git tag "v1.6.40" (see <https://github.com/glennrp/libpng/releases/tag/v1.6.40>)

You can add a filter on your new url to remove that starting v if it's not part of the url

**Comment by darcy-shen** [comment]
> it's not about opinion, xmake uses the version name as git tags if archive download fails.

Let me find out why #2264 works fine.

**Comment by SirLynix** [comment]
> > it's not about opinion, xmake uses the version name as git tags if archive download fails.
> 
> Let me find out why #2264 works fine.

Because it successfully downloaded the archive (and thus didn't test the git tag), but you can see on this CI that xmake tries to use the version as a git name:
```
  => download https://github.com/glennrp/libpng/archive/refs/tags/v1.6.40.tar.gz .. failed
/usr/bin/git clone https://github.com/glennrp/libpng.git -c core.fsmonitor=false source.tmp/libpng
Cloning into 'source.tmp/libpng'...
/usr/bin/git -c core.fsmonitor=false checkout 1.6.40
error: pathspec '1.6.40' did not match any file(s) known to git
```

**Comment by darcy-shen** [comment]
I think it is because I have not update the hash.

**Comment by SirLynix** [comment]
> I think it is because I have not update the hash.

Yes, the hash is wrong so it tried to fallback on git, and failed because you removed the starting "v".

You should put that starting `v` back and update hash, same for autodiff

**Comment by darcy-shen** [comment]
> autodiff url/version naming is wrong and should be updated

You are right! It is not version, but tag.

File: packages/l/libpng/xmake.lua [hunk 1]
```diff
@@ -3,13 +3,13 @@ package("libpng")
...
-    add_versions("1.6.40", "ab3f88779f0661bbb07c60e778fda782216bff70355d86848fbf6a327084563a")
-    add_versions("1.6.37", "c2c50c13a727af73ecd3fc0167d78592cf5e0bca9611058ca414b6493339c784")
-    add_versions("1.6.36", "6274d3f761cc80f7f6e2cde6c07bed10c00bc4ddd24c4f86e25eb51affa1664d")
-    add_versions("1.6.35", "3d22d46c566b1761a0e15ea397589b3a5f36ac09b7c785382e6470156c04247f")
-    add_versions("1.6.34", "7ffa5eb8f9f3ed23cf107042e5fec28699718916668bbce48b968600475208d3")
+    add_versions("v1.6.40", "62d25af25e636454b005c93cae51ddcd5383c40fa14aa3dae8f6576feb5692c2")
+    add_versions("v1.6.37", "ca74a0dace179a8422187671aee97dd3892b53e168627145271cad5b5ac81307")
+    add_versions("v1.6.36", "5bef5a850a9255365a2dc344671b7e9ef810de491bd479c2506ac3c337e2d84f")
+    add_versions("v1.6.35", "6d59d6a154ccbb772ec11772cb8f8beb0d382b61e7ccc62435bf7311c9f4b210")
+    add_versions("vs1.6.34", "e45ce5f68b1d80e2cb9a2b601605b374bdf51e1798ef1c2c2bd62131dfcf9eef")
```
Comment (SirLynix) [inline packages/l/libpng/xmake.lua:12]: vs => v

**Review by SirLynix — APPROVED** [review]

**Pull request merged**